File: algorithms/0119.pascals-triangle-ii/solution.rb
# @param {Integer} row_index
# @return {Integer[]}
def get_row(row_index)
  rows = [[1]]

  # generate
  1.upto(row_index) do |i|
    # init row
    rows << [1] * (i + 1)

    # calculation
    (1...i).each do |j|
      rows[i][j] = rows[i - 1][j - 1] + rows[i - 1][j]
    end
  end

  rows[-1]
end

File: algorithms/0119.pascals-triangle-ii/solution.rs
impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let mut rows: Vec<Vec<i32>> = vec![vec![1]];

        for i in 1..=row_index as usize {
            // init row
            rows.push(vec![1; i + 1]);

            // calculation
            for j in 1..i {
                rows[i][j] = rows[i - 1][j - 1] + rows[i - 1][j];
            }
        }

        rows.last().unwrap().to_vec()
    }
}
